### PR TITLE
chore: renamed react-native-codegen to @react-native/codegen

### PR DESCRIPTION
**Pull request by shivenmian**
## Summary

Renamed react-native-codegen package to @react-native/codegen and updated references, without changing the folder name; part of RFC480 (#34692). Follow-up from https://github.com/facebook/react-native/pull/34578

## Changelog

[General] [Changed] - Renamed react-native-codegen package to @react-native/codegen and updated references

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 - Accept TypeScript type `T | null | undefined` as a maybe type of T in turbo module. ([9ecd203eec](https://github.com/facebook/react-native/commit/9ecd203eec97e7d21d10311d950c9f8f30c7a4b1) by [@ZihanChen-MSFT](https://github.com/ZihanChen-MSFT))
 - Bump react-native-gradle-plugin to 0.70.3 ([e33633644c](https://github.com/facebook/react-native/commit/e33633644c70ea39af6e450fcf31d9458051fd5f) by [@dmytrorykun](https://github.com/dmytrorykun))
-- Bump react-native-codegen to 0.70.5 ([6a8c38eef2](https://github.com/facebook/react-native/commit/6a8c38eef272e79e52a35941afa9c3fe9e8fc191) by [@dmytrorykun](https://github.com/dmytrorykun))
+- Bump @react-native/codegen to 0.70.5 ([6a8c38eef2](https://github.com/facebook/react-native/commit/6a8c38eef272e79e52a35941afa9c3fe9e8fc191) by [@dmytrorykun](https://github.com/dmytrorykun))
```
Comment (cortinico) [inline CHANGELOG.md:14]: Let's amend changes to the CHANGELOG 👍

Comment (shivenmian) [reply CHANGELOG.md]: @cortinico done

**Comment by shivenmian** [comment]
@cortinico done. I'm not seeing the `test_js` task btw - CircleCI isn't running. Is there something more I need to do for that?

File: .circleci/verdaccio.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ packages:
   '@react-native-community/eslint-plugin':
     access: $all
     publish: $all
-  'react-native-codegen':
+  '@react-native/codegen':
     access: $all
     publish: $all
```
Comment (cortinico) [inline .circleci/verdaccio.yml:31]: I think we can just remove this section entierly

**Comment by cortinico** [comment]
> something more I need to do for that?

Let's try to rebase. If it's still not working, I'll try to look more closely into this

**Comment by shivenmian** [comment]
@cortinico done

**Comment by cortinico** [comment]
Hey @shivenmian, I've rebased your PR as the CI was not running.
Now `eslint` is failing on CI. Are you able to take a look at it?

**Comment by shivenmian** [comment]
@cortinico should work now

**Comment by shivenmian** [comment]
@cortinico looks like the JS loop passed. It doesn't seem like the Android / Windows loop failures are related to this, and the iOS one seems like it can't find `@react-native/codegen` from npm - this should be fixed once the package change makes it way up to npm as well?

**Comment by shivenmian** [comment]
@cortinico this good to go?

**Comment by cortinico** [comment]
> @cortinico this good to go?

Sorry I was off some days and missed this ping.
The PR seems good to go. I'm investigating if we can use Verdaccio also for the template tests similarly to what we did here https://github.com/facebook/react-native/pull/34577

Is this something you would be interested in looking into setting up for the whole React Native repo (as this would unblock a lot of other PRs like this one).

**Comment by shivenmian** [comment]
Sure. Quick question:

> I'm investigating if we can use Verdaccio also for the template tests similarly to what we did here #34577

By "template tests", do you mean the tests that are failing right now on this PR (i.e test_android_template_... and test_ios_template_...)?

**Comment by cortinico** [comment]
> By "template tests", do you mean the tests that are failing right now on this PR (i.e test_android_template_... and test_ios_template_...)?

Yes exactly 👍

**Comment by shivenmian** [comment]
Sure - I'll look at it this week.

**Comment by hoxyq** [comment]
Hey @shivenmian, thanks for contributing. To make this work (both internally and in open-source) I need to add some changes. Could you please share if this is okay for you?

**Comment by shivenmian** [comment]
@hoxyq yep go ahead!

**Pull request closed without merging**